l3_dump.py

```diff
@@ -8,18 +8,67 @@
 import sys
 import struct
 import subprocess
+import os
 
 if len(sys.argv) != 3:
     print(f"Usage: {sys.argv[0]} <logfile> <program-binary>")
+    print(" ")
+    print("Note: If L3 logging is done for <program-binary> with L3_LOC_ENABLED=1")
+    print("we expect to find the LOC-decoder binary named <program-binary>_loc")
     sys.exit(0)
 
+# #############################################################################
+# Check for required LOC-decoder binary, based on environment of run
+# If L3_LOC_ENABLED=1 is set in the env, search for a LOC-decoder binary
+# named "<program_binary>_loc".
+# #############################################################################
+PROGRAM_BIN = sys.argv[2]
+DECODE_LOC_ID = 0   # By default, we expect L3 logging was done w/LOC OFF.
+LOC_ENABLED = "L3_LOC_ENABLED"
+LOC_DECODER = "LOC_DECODER"
+if LOC_ENABLED in os.environ:
+    env_var_val = os.getenv(LOC_ENABLED)
+    if env_var_val == "1":
+        LOC_DECODER = PROGRAM_BIN + "_loc"
+        if os.path.exists(LOC_DECODER) is False:
+            print(f"Env-var {LOC_ENABLED}=1 is set, but required "
+                  + f"LOC-decoder binary {LOC_DECODER} is not found.")
```

Comment:
There is one invocation of this tool in build.yml to exercise this error-check condition.